static/src/javascripts/projects/common/modules/navigation/search.js

```diff
@@ -0,0 +1,169 @@
+// @flow
+
+import bean from 'bean';
+import fastdom from 'fastdom';
+import $ from 'lib/$';
+import config from 'lib/config';
+import throttle from 'lodash/functions/throttle';
+
+const focusSearchField = (): void => {
+    const $input = $('input.gsc-input');
+    if ($input.length > 0) {
+        $input.focus();
+    }
+};
+
+class Search {
+    gcsUrl: string;
+    resultSetSize: number;
+
+    constructor(): void {
+        const toggle = document.querySelector('.js-search-toggle');
+        let searchLoader;
+
+        if (
+            config.switches.googleSearch &&
```

Comment:
We recommend using `config.get` to access properties of config. eg. `config.get('switches.googleSearch')`. `config.get` gracefully handles attempts to reference a property of config that doesn't exist in the config. As explained here https://github.com/guardian/frontend/pull/17728